### ISSUE TITLE
Remove duplicate code on recipes

### ISSUE THREAD

**Issue by faustinoaq** [amberframework/recipes#16]
Hi :sweat_smile: 

Currently `--recipe` flag is working very nice on master (not released yet), also the recipes are being automatically tested and deployed :tada: https://github.com/amberframework/recipes/releases

Although, this repo has some duplicated code across recipes, we have some options to fix this:

1. Use symlinks to reuse existent code/directories/files, before deploy the symlinks should be replaced by the real folder/file, once deployed and zipped the recipe should not contain symlinks.

2. Use multiple liquid layouts to reuse templates. We would need a common layout directory, so every recipe can include it and reuse the code.

3. Use one recipe per repository this won't remove the duplicated code, but just move the code to a new repo for each recipe. This would work by using `shards install`, although the maintainability is not easy, because we will require to test, maintain, deploy and review issue/PRs for all recipe repositories..

/cc @drujensen @damianham @amberframework/contributors @amberframework/core-team

**Comment by faustinoaq** [comment]
Please vote using

1. :smile: (option 1)
2. :tada: (option 2)
3. :heart: (option 3)

**Comment by faustinoaq** [comment]
> hierarchical recipes

@damianham Oh yeah! I added your option, and I think is a very nice idea :+1:

**Comment by drujensen** [comment]
@damianham Thanks for looking into this.

**Comment by drujensen** [comment]
Would be nice `shards install drujensen/basic_granite` existed where it would add this to your shards.yml and install it.  https://github.com/crystal-lang/shards/blob/master/src/commands/install.cr

But there is still a chicken/egg thing since the `shards.yml` won't exist until the project templates are ran.  It would require a bootstrap step which is probably what you are already doing.  

I still think it would be nice to be able to update the recipes by just updating the `shard.yml`.

**Comment by faustinoaq** [comment]
@drujensen I think `shards install` doesn't support that yet :sweat_smile: 

See: [install cli](https://github.com/crystal-lang/shards/blob/master/src/cli.cr#L49)

See: https://github.com/crystal-lang/shards/issues/144

**Comment by damianham** [comment]
@drujensen yes there is a chicken/egg situation and I am trying to keep it as simple as possible.  I will see if I can find a method to specify the recipe in the app/shard.yml and also bootstrap the app from the recipe.
